include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -319,3 +319,27 @@ def MuxOp : RTLOp<"mux",
     }]>
   ];
 }
+
+def KnownBitWidthType : Type<CPred<[{getBitWidth($_self) != -1}]>,
+  "Type wherein the bitwidth in hardware is known">;
+
+def BitcastOp: RTLOp<"bitcast", [NoSideEffect]> {
+  let summary = [{
+    Cast one value to to another value of the same size and
+    potentially different type, using a bitwise reinterpretation of the
+    value.
+  }];
+
+  let arguments = (ins KnownBitWidthType:$from);
```

Comment:
I think that `$input` is more conventional in the RTL dialect for unary ops like this.  I'd also recommend `$result` for the result name.